internal/resources/google/container_registry.go

```diff
@@ -0,0 +1,57 @@
+package google
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/infracost/infracost/internal/usage"
+)
+
+type ContainerRegistry struct {
+	Address                     string
+	Region                      string
+	Location                    string
+	StorageClass                string
+	StorageGB                   *float64                             `infracost_usage:"storage_gb"`
+	MonthlyClassAOperations     *int64                               `infracost_usage:"monthly_class_a_operations"`
+	MonthlyClassBOperations     *int64                               `infracost_usage:"monthly_class_b_operations"`
+	MonthlyEgressDataTransferGB *ContainerRegistryNetworkEgressUsage `infracost_usage:"monthly_egress_data_transfer_gb"`
+}
+
+var ContainerRegistryUsageSchema = []*schema.UsageItem{
+	{Key: "storage_gb", ValueType: schema.Float64, DefaultValue: 0},
+	{Key: "monthly_class_a_operations", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "monthly_class_b_operations", ValueType: schema.Int64, DefaultValue: 0},
+	{Key: "monthly_data_retrieval_gb", ValueType: schema.Float64, DefaultValue: 0},
+	{
+		Key:          "monthly_egress_data_transfer_gb",
+		ValueType:    schema.SubResourceUsage,
+		DefaultValue: &usage.ResourceUsage{Name: "monthly_egress_data_transfer_gb", Items: ContainerRegistryNetworkEgressUsageSchema},
+	},
+}
+
+func (r *ContainerRegistry) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+	if r.MonthlyEgressDataTransferGB == nil {
+		r.MonthlyEgressDataTransferGB = &ContainerRegistryNetworkEgressUsage{}
+	}
+}
+
+func (r *ContainerRegistry) BuildResource() *schema.Resource {
+	region := r.Region
+	components := []*schema.CostComponent{
+		dataStorage(r.Location, r.StorageClass, r.StorageGB),
+	}
+
+	components = append(components, operations(r.StorageClass, r.MonthlyClassAOperations, r.MonthlyClassBOperations)...)
+
+	r.MonthlyEgressDataTransferGB.Region = region
```

Comment:
Should we also check `r. MonthlyEgressDataTransferGB ` for nil here? Maybe we should move the nil check from above to here to its in the same function that we access it.

Reply:
same as above